molecule/default/externally-managed-python.yml

```
@@ -0,0 +1,47 @@
+---
+# This is in place to restore a destructive action in geerlingguy's Ansible
+# Docker images that we use for testing. The change is fine for the intended
+# purpose of the images but not for how we use them.
+- name: Ensure Python is marked as externally managed if appropriate
+  hosts: all
+  become: true
+  become_method: ansible.builtin.sudo
+  tasks:
+    - name: Ensure Python is marked as externally managed
+      when:
+        - ansible_distribution in ["Debian", "Ubuntu"]
+        - ansible_distribution_release not in ["bullseye", "buster", "focal", "jammy"]
+      block:
+        - name: Gather package facts
+          ansible.builtin.package_facts:
+            manager: auto
+
+        - name: Ensure the EXTERNALLY-MANAGED file is present if Python 3 is installed
+          when: '"python3" in ansible_facts.packages'
+          block:
+            # This gets a list of unique list of installed Python packages in the form of major.minor
+            # by taking the list of installed Python packages and:
+            # 1. Extracting the version from each package's information
+            # 2. Removing any version information after the major.minor version
+            # 3. Ensuring there are no duplicates
+            #
+            # NOTE:
+            # The regex expressions used in the regex_replace filter must use double backslashes if
+            # the value of python_versions is changed from a multiline string. This is due to how the
+            # YAML is processed when Ansible reads the playbook.
```

Can this comment be reworded to mention that single backslashes are used in the regex?  It's a little confusing- I eventually figured out what you meant, but initially, I was expecting to see double backslashes in the regex.

Here's my shot at it, but feel free to edit:
```suggestion
            # The value of python_versions is a multiline string so that we can use single backslashes
            # in the regex expressions in the regex_replace filter. If python_versions was not a multiline
            # string, we would have to use double backslashes. This is due to how the YAML is
            # processed when Ansible reads the playbook.
```

---

I agree that @dav3r's language is easier to parse.

---

Thoughts about this?

```suggestion
            # The value of python_versions is a multiline string so the regex expressions used in the
            # regex_replace filter must use single backslashes for special sequences. If the value of
            # python_versions is changed from a multiline string the special sequences must be
            # modified to use double backslashes instead. This is due to how the YAML is processed
            # when Ansible reads the playbook.
```

@dav3r's suggestion implies that we chose to use a multiline string to use single backslashes when it's more an implementation detail. This is a multiline string to make the chaining filters easier to parse.

---

I'm good with that after some small tweaks:
```suggestion
            # Since the value of python_versions is a multiline string, the regex expressions used in
            # the regex_replace filter can use single backslashes for special sequences. If the value
            # of python_versions were to be changed from a multiline string, the special sequences
            # would have to be modified to use double backslashes instead. This is due to how the
            # YAML is processed when Ansible reads the playbook.
```

---

@dav3r Problem again here. It's not _can be used_ it is _must be used_. The regex expressions will _only_ work as expected if you use single backslashes (let's not talk about how much banging my head against the wall was done before finding out this fact).

---

Got it, so then if we change `can` to `must`, we are all good here?
```suggestion
            # Since the value of python_versions is a multiline string, the regex expressions used in
            # the regex_replace filter must use single backslashes for special sequences. If the value
            # of python_versions were to be changed from a multiline string, the special sequences
            # must be modified to use double backslashes instead. This is due to how the YAML is
            # processed when Ansible reads the playbook.
```